### PR TITLE
refactor(KSA VAT): QR Code as per ZATKA specification

### PR DESCRIPTION
**Pull request by ahmadpak**
Verified using the following app:
Play store
https://play.google.com/store/apps/details?id=com.posbankbh.einvoiceqrreader

App store
https://apps.apple.com/sa/app/e-invoice-qr-reader-ksa/id1580793042

image

### CODE REVIEW

File: erpnext/regional/saudi_arabia/utils.py [hunk 1]
```diff
@@ -28,24 +31,76 @@ def create_qr_code(doc, method):
...
+			# 	doc.company,
+			# 	'company_name_in_arabic')
+
+			# if not seller_name:
+			# 	frappe.throw(_('Arabic name missing for {} in the company document'.format(doc.company)))
+
+			seller_name = doc.company
+			tag = bytes([1]).hex()
+			length = bytes([len(seller_name)]).hex()
+			value = seller_name.encode('utf-8').hex()
```
Comment (ahmadpak) [inline erpnext/regional/saudi_arabia/utils.py:54]: @deepeshgarg007 This is done as per the mentioned specification in the docs but somehow does not convert Arabic name to proper hex code.

Comment (nesscale-com) [reply erpnext/regional/saudi_arabia/utils.py:56]: @ahmadpak I think need to encode string before length function. 
			length = bytes([len(seller_name.encode('utf-8'))]).hex()

Comment (bhavesh95863) [reply erpnext/regional/saudi_arabia/utils.py:54]: @ahmadpak I think needs to encode string before length.
			length = bytes([len(seller_name.encode('utf-8'))]).hex()

**Comment by deepeshgarg007** [comment]
Merged via https://github.com/frappe/erpnext/pull/28605. Closing this PR

**Comment by shiraski** [comment]
ho w to update this qr code my existing system

**Pull request closed without merging**